esphome/components/partition/light.py

```diff
@@ -63,6 +84,18 @@ def validate_from_to(value):
     }
 )
 
+FINAL_VALIDATE_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_SEGMENTS): cv.ensure_list(
+            cv.All(
+                cv.Any(ADDRESSABLE_SEGMENT_SCHEMA, NONADDRESSABLE_SEGMENT_SCHEMA),
+                validate_segment,
+            )
+        ),
```

Comment:
```suggestion
        cv.Required(CONF_SEGMENTS): [validate_segment],
```

When final validate is called the main validation has already passed. So we can depend on that then